### PR TITLE
Close milestones after release

### PR DESCRIPTION
**Pull request by mathias-baumann-sociomantic**

### CODE REVIEW

**Comment by mathias-baumann-sociomantic** [comment]
Rebased on current master

File: src/release/main.d [hunk 1]
```diff
@@ -145,6 +145,29 @@ void main ( string[] params )
...
+                 open_milestones.map!(a=>a.title));
+
+        if (readYesNoResponse("Would you like to close them now?"))
+        {
+            foreach (milestone; open_milestones)
+            {
+                writefln("Closing %s .. (%s open issues)",
+                         milestone.title, milestone.open_issues);
+
+                updateMilestoneState(con, repo, milestone.number, State.closed);
```
Comment (leandro-lucarella-sociomantic) [inline src/release/main.d]: Shouldn't this be just another item on the TODO list? Why asking for confirmation separately (IIRC on how the tool works, I haven't used it lately).

File: src/release/main.d [hunk 2]
```diff
@@ -159,19 +159,58 @@ void main ( string[] params )
...
+        dg = delegate to call until it succeeds
+
+*******************************************************************************/
+
+void keepTrying ( void delegate ( ) dg )
+{
+    import core.thread;
+    import std.stdio;
+
+    while (true) try
```
Comment (leandro-lucarella-sociomantic) [inline src/release/main.d]: I would have a finite number of retries. Maybe ask the if you should keep trying after a few tries, but going infinite seems a bit extreme.

File: src/release/main.d [hunk 3]
```diff
@@ -159,19 +159,58 @@ void main ( string[] params )
...
+        dg();
+
+        stdout.flush();
+        break;
+    }
+    catch (Exception exc)
+    {
+        writefln("failed: %s", exc.msg);
+        writefln("retrying ...");
+        Thread.sleep(1.seconds);
```
Comment (leandro-lucarella-sociomantic) [inline src/release/main.d]: Also a ["human" exponential backoff](https://github.com/sociomantic-tsunami/git-hub/blame/2b1f5329c41480ccd9599779faff23bda736e163/git-hub#L863-L876) could be a good idea :P

Comment (mathias-baumann-sociomantic) [inline src/release/main.d]: From my experience it's working after the second attempt 99% of the cases..

**Comment by mathias-baumann-sociomantic** [comment]
fixed copy right

File: src/release/main.d (hunk 1 above)
Comment (mathias-baumann-sociomantic) [inline src/release/main.d]: I don't understand what you are asking.. what TODO list...

File: src/release/main.d (hunk 2 above)
Comment (leandro-lucarella-sociomantic) [inline src/release/main.d]: Still, IMHO is better to have a limit, but I won't fight for it if you prefer it this way.

File: src/release/main.d (hunk 1 above)
Comment (mathias-baumann-sociomantic) [inline src/release/main.d]: Well, internally it is not one of those actions yet. I suppose it make sense to turn it into one, but that would be something for another PR

File: src/release/main.d (hunk 2 above)
Comment (mathias-baumann-sociomantic) [inline src/release/main.d]: Implemented a limit of 10 now

**Comment by mathias-baumann-sociomantic** [comment]
updated, implemented limit of 10 retry attempts

File: src/release/main.d [hunk 4]
```diff
@@ -609,3 +611,28 @@ string getUpstream ( )
...
+
+
+void sanityCheckMilestone ( ref HTTPConnection con, ref Repository repo, string name )
+{
+    import release.github;
+    import std.algorithm;
+    import std.stdio;
+    import std.range;
+
+    auto mstone_list = listMilestones(con, repo).find!(a=>a.title==name);
```
Comment (nemanja-boric-sociomantic) [inline src/release/main.d]: is this spacing (no spaces around `==`) consistent with the project?

File: src/release/main.d [hunk 5]
```diff
@@ -159,19 +159,74 @@ void main ( string[] params )
...
         }
     }
 
     writefln("All done.");
 }
 
 
+/*******************************************************************************
+
+    Repeatatly tries to call dg() if it threw an exception, while also keeping
```
Comment (nemanja-boric-sociomantic) [inline src/release/main.d]: typo `repeatatly`

File: src/release/main.d [hunk 6]
```diff
@@ -138,12 +138,12 @@ void main ( string[] params )
...
-    {
-        import octod.api.releases;
-
-        writefln("Creating %s ...", ver);
-        con.createRelease(repo, ver, ver, getTagMessage(ver));
-    }
+        keepTrying(
+        {
+            import octod.api.releases;
+            writef("Creating %s ... ", ver);
```
Comment (nemanja-boric-sociomantic) [inline src/release/main.d]: `writefln`?

**Comment by mathias-baumann-sociomantic** [comment]
Pushed update
